src/io/fld-read.jl

```diff
@@ -1,297 +1,26 @@
-
-		filetype = arg_get(header, "filetype", false)
-		chat && @info("Current file = '$file', External file = '$extfile', type='$filetype'")
-
-		_skip = occursin("skip=",prod(header)) ?
-
-    	arg_get([prod(header)], "skip", false) : 0
-
-		if filetype == "ascii"
-			tmp = dirname(file)
-			extfile = joinpath(tmp, extfile)
-			chat && @info("extfile = $extfile")
-			isfile(extfile) || throw("no ascii file $extfile")
-			format, _, _ = datatype_fld_to_mat(datatype)
-			return fld_read_ascii(extfile, (dims...,), format)
-
-		elseif filetype != "multi"
-			if !isfile(extfile)
-				fdir = file
-				slash = findlast(isequal('/'), fdir) # todo: windows?
-				isnothing(slash) && throw("cannot find external file $extfile")
-				fdir = fdir[1:slash]
-				extfile = fdir * extfile # add directory
-				!isfile(extfile) && throw("no external ref file $extfile")
-			end
-		else
-			throw("multi not supported yet") # todo
-		end
-	else
-		filetype = ""
-		extfile = ""
-	end
-
-	# finally, read the binary data
-	format, endian, bytes = datatype_fld_to_mat(datatype)
-
-	# single file reading
-	data = fld_read_single(file, fid, dims, datatype, fieldtype,
-		is_external_file, extfile, format, endian, bytes, _skip)
-
-	close(fid)
-
-	return data
-end
-
-
-
-# todo: currently supports only one entry per line (see fld_read.m)
-function fld_read_ascii(extfile::AbstractString, dims::Dims, datatype::DataType)
-	data = zeros(datatype, dims)
-	open(extfile, "r") do fid
-		for i in 1:length(data)
-			tmp = readline(fid)
-			data[i] = parse(Float64, tmp)
-		end
-	end
-	return data
-end
-
-
-function fld_read_single(
-	file, fid, dims, datatype, fieldtype,
-	is_external_file, extfile, format::DataType, endian, bytes, _skip,
-)
-
-	# reopen file to same position, with appropriate endian too.
-	if is_external_file
-		fid = open(extfile)
-	end
-
-	skip(fid,_skip)
-
-	rdims = dims # from handling slice option
-
-	# read binary data and reshape appropriately
-	data = Array{format}(undef, rdims...)
-	try
-		read!(fid,data)
-	catch
-		@info("rdims=$rdims")
-		throw("file count != data count")
-	end
-
-	return endian === :le ? htol.(data) :
-		endian === :be ? hton.(data) :
-		throw("bug $endian")
-end
-
+    fld_header()
 
+Deprecated: use `AVSfldIO.fld_header` instead.
 """
-`header = string_to_array(header_lines)`
-
-convert long string with embedded newlines into string array
-"""
-function string_to_array(header_lines::String)
-	newline = '\n'
-
-	# ensure there is a newline at end, since dumb editors can forget...
-	if header_lines[end] != newline
-		header_lines *= newline
-	end
-
-	ends = findall(isequal(newline), header_lines)
-	(length(ends) <= 0) && throw("no newlines?")
-
-	header = split(header_lines, newline, keepempty=false)
-
-	# strip comments (lines that begin with #)
-	header = filter(line -> line[1] != '#', header)
-
-	return header
+function fld_header()
+    throw("fld_header is deprecated: use AVSfldIO.fld_header instead.")
 end
```

Comment:
```suggestion
@deprecate fld_header AVSfldIO.fld_header
```

Reply:
Thanks very much for this suggestion.  I didn't know about this command!
I will use it in the next commit (before merging this PR).

It's a bit sad though that by default there is no warning given to users:
https://docs.julialang.org/en/v1/base/base/#Base.@deprecate
So it seems like most users will not know about this change until we really remove it anyway.

Reply:
> So it seems like most users will not know about this change until we really remove it anyway.

Correct... This is because Julia is designed for high performance, and warnings will hurt the performance. Another reason that people need to be careful about the semantics versioning 😄